testutils/utils.go

```diff
@@ -40,23 +40,25 @@ func MakeTempDir(t testing.TB, tempDir string) func() error {
 // RegisterConsulService regsiters a service to the Consul Catalog. The Consul
 // sdk/testutil package currently does not support a method to register multiple
 // service instances, distinguished by their IDs.
-func RegisterConsulService(t *testing.T, srv *testutil.TestServer,
+func RegisterConsulService(tb testing.TB, srv *testutil.TestServer,
 	s testutil.TestService, health string) {
 
 	var body bytes.Buffer
 	enc := json.NewEncoder(&body)
-	require.NoError(t, enc.Encode(&s))
+	require.NoError(tb, enc.Encode(&s))
 
 	u := fmt.Sprintf("http://%s/v1/agent/service/register", srv.HTTPAddr)
-	resp := RequestHTTP(t, http.MethodPut, u, body.String())
+	resp := RequestHTTP(tb, http.MethodPut, u, body.String())
 	defer resp.Body.Close()
 
-	srv.AddCheck(t, s.ID, s.ID, testutil.HealthPassing)
+	if t, ok := tb.(*testing.T); ok {
+		srv.AddCheck(t, s.ID, s.ID, testutil.HealthPassing)
```

Comment:
In my branch I've pulled in AddCheck and related code into testutils/sdk/, changing as little as possible to get them working with the idea that it can be ripped out after the next consul/sdk release. 

I'll update this to use it as well once merged as you should be merging first.

Reply:
Sounds good, thanks for wrangling those updates!